### DISCUSSION TITLE
Search on binary vector failing

### DISCUSSION THREAD

**Discussion by dee-corr** [milvus-io/milvus#19033] [Q&A and General discussion] [answered]
Hello,

Hello, I'm new to Milvus. I'm trying to do a search using Tanimoto on a collection of 4096 dim binary vectors and getting the below error. The collection has approx 24,000,000 records.

<img width="1487" alt="image" src="https://user-images.githubusercontent.com/51026617/188520412-471f75ba-2cd5-4c8d-af75-12d67a2b884c.png">

Am I doing something wrong?

**Comment by yhmo** [comment, marked as answer]
The "metric_type" should be "TANIMOTO", not "BIN_IVF_FLAT".

Here is an example:

```
if __name__ == "__main__":
    connections.connect()
    print("========== BINARY VECTOR ============")
    dim = 4096
    binary_fields = [
        FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
        FieldSchema(name="bin_vec", dtype=DataType.BINARY_VECTOR, dim=dim)
    ]
    schema = CollectionSchema(fields=binary_fields)
    collection = Collection("AAA", schema, consistency_level="Strong")
    print("collection created")

    num = 2000
    binary_vectors = []
    for i in range(num):
        raw_vector = [random.randint(0, 1) for i in range(dim)]
        binary_vectors.append(bytes(np.packbits(raw_vector, axis=-1).tolist()))
    entities = [
        binary_vectors
    ]

    for i in range(1):
        t1 = time.time()
        print("Start insert")
        collection.insert(entities)
        t2 = time.time()
        print("Insert", num, "entities time cost", (t2-t1), "seconds")

    print("Row count:", collection.num_entities)

    index_param = {"index_type": "BIN_IVF_FLAT", "metric_type": "TANIMOTO", "params": {"nlist": 128}}
    collection.create_index(field_name="bin_vec", index_params=index_param)

    collection.load()

    search_params = {
        "metric_type": "TANIMOTO",
        "params": {"nprobe": 10},
    }
    result = collection.search(data=binary_vectors[:1], anns_field="bin_vec", param=search_params, limit=3)
    for hits in result:
        for hit in hits:
            print(f"hit: {hit}")
```

**Comment by yhmo** [comment]
In v2.1.1, we made an improvement that returns the error message with more details about the error. In this case, if you set the "metric_type" to be "TANIMOTO", the error message is:

pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=2, reason=Search 2 failed, reason [UnexpectedError] Assert "field_indexing->metric_type_ == search_info.metric_type_" at /home/yhmo/git-go/milvus/internal/core/src/query/SearchOnSealed.cpp:43
 => **Metric type of field index isn't the same with search info** err %!w(<nil>))>

**Comment by dee-corr** [comment]
Thank you @yhmo !!
